### PR TITLE
[CDEC-322] Move crypto orphan instances to correct places

### PR DESCRIPTION
**Pull request by ruhatch**
## Description

Orphan instances for `From/ToJSON`, `Binary` and `AsBinary` were all in their own files. They have now been moved to be with their types and any external orphan instances have been collected in `Pos.Crypto.Orphans`. This is a big win because now there are no `Pos.Binary.Crypto` or `Pos.Aeson.Crypto` instances anywhere in the codebase - the instances are brought along with the types!

## Linked issue

[CDEC-322](https://iohk.myjetbrains.com/youtrack/issue/CDEC-322)

## Type of change

- [~] 🐞 Bug fix (non-breaking change which fixes an issue)
- [~] 🛠 New feature (non-breaking change which adds functionality)
- [~] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)
- [x] 🏭 Refactoring that does not change existing functionality but does improve things like code readability, structure etc
- [~] 🔨 New or improved tests for existing code
- [~] ⛑ git-flow chore (backport, hotfix, etc)

## Developer checklist

- [x] I have read the [style guide](https://github.com/input-output-hk/cardano-sl/blob/develop/docs/style-guide.md) document, and my code follows the code style of this project.
- [x] If my code deals with exceptions, it follows the [guidelines](https://github.com/input-output-hk/cardano-sl/blob/develop/docs/exceptions.md).
- [x] I have updated any documentation accordingly, if needed. Documentation changes can be reflected in opening a PR on [cardanodocs.com](https://github.com/input-output-hk/cardanodocs.com), amending the inline [Haddock](https://www.haskell.org/haddock/) comments, any relevant README file or one of the document listed in the [docs](https://github.com/input-output-hk/cardano-sl/tree/develop/docs) directory.

## Testing checklist

- [~] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by erikd** [comment]
Once that whitespace thing is sorted I'm happy with this, but since this is the crypto package I'd like @vincenthz to review it was well.

File: crypto/Pos/Crypto/HD.hs [hunk 1]
```diff
@@ -136,7 +148,7 @@ unpackHDAddressAttr (HDPassphrase passphrase) (HDAddressPayload payload) = do
     case unpackCF of
         Left _ -> Nothing
         Right p -> case decodeFull' p of
-            Left _ -> Nothing
+            Left _     -> Nothing
```
[the remark this answers is not in the document]
Comment (erikd) [reply crypto/Pos/Crypto/HD.hs]: Chatted to @ruhatch and it seems this was `sytlish-haskell`with the cardano-sl setting file, so we'll let this go.

[the remark this answers is not in the document]
Comment (ruhatch) [reply crypto/Pos/Crypto/HD.hs]: This is from stylish-haskell conforming to the repo's `.stylish-haskell.yaml`

**Comment by vincenthz** [comment]
this really need to be split in much smaller logical unit for review, and considering there's no tests covering the actual serialization of binary nor json, this seems like a complicated review exercise with limited safety.

File: crypto/Pos/Crypto/AsBinary.hs [hunk 2]
```diff
@@ -1,77 +0,0 @@
...
-    build _ = "secret \\_(o.o)_/"
-
-instance Buildable (AsBinary DecShare) where
-    build _ = "share \\_(*.*)_/"
-
-instance Buildable (AsBinary EncShare) where
-    build _ = "encrypted share \\_(0.0)_/"
-
-instance Bi (AsBinary VssPublicKey) => Buildable (AsBinary VssPublicKey) where
-    build = bprint ("vsspub:"%shortHashF) . hash
```
Comment (mhuesch) [inline crypto/Pos/Crypto/AsBinary.hs]: Moved to `Pos.Crypto.SecretSharing`

File: crypto/Pos/Crypto/Orphans.hs [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+       (
+       ) where
+
+import           Prelude (show)
+import           Universum hiding (show)
+
+import qualified Cardano.Crypto.Wallet as CC
+import qualified Crypto.SCRAPE as Scrape
+import           Crypto.Scrypt (EncryptedPass (..))
+import qualified Crypto.Sign.Ed25519 as Ed25519
```
Comment (mhuesch) [inline crypto/Pos/Crypto/Orphans.hs]: 👍 better name

File: crypto/Pos/Binary/Crypto.hs [hunk 4]
```diff
@@ -1,271 +0,0 @@
...
-    encode (EdStandard.SecretKey k) = encode k
-    decode = EdStandard.SecretKey <$> decode
-
-instance Bi EdStandard.Signature where
-    encode (EdStandard.Signature s) = encode s
-    decode = EdStandard.Signature <$> decode
-
-deriving instance Bi RedeemPublicKey
-deriving instance Bi RedeemSecretKey
-deriving instance Typeable a => Bi (RedeemSignature a)
```
Comment (mhuesch) [inline crypto/Pos/Binary/Crypto.hs]: Instances moved to where the types live, or to `Pos.Crypto.Orphans`.

**Pull request merged**